mir/src/passes/constant_propagation.rs

```diff
@@ -121,6 +120,185 @@ impl ConstantPropagation<'_> {
             try_fold_const_binary_op(lhs, rhs, exp.clone(), exp_ref.span())
         }
     }
+
+    fn visit_accessor_bis(
+        &mut self,
+        _graph: &mut Graph,
+        accessor: Link<Op>,
+    ) -> Result<Option<Link<Op>>, CompileError> {
+        let mut updated_accessor = None;
+
+        {
+            let accessor_ref = accessor.as_accessor().unwrap();
+            let indexable = accessor_ref.indexable.clone();
+            let mir_access_type = accessor_ref.access_type.clone();
+            let offset = accessor_ref.offset;
+
+            match mir_access_type {
+                MirAccessType::Default => {
+                    updated_accessor = Some(indexable.clone());
+
+                    if let Some(value) = indexable.clone().as_value() {
+                        let mir_value = value.value.value.clone();
+
+                        if let MirValue::TraceAccess(trace_access) = mir_value {
+                            let new_node = Value::create(SpannedMirValue {
+                                span: value.value.span(),
+                                value: MirValue::TraceAccess(TraceAccess {
+                                    segment: trace_access.segment,
+                                    column: trace_access.column,
+                                    row_offset: trace_access.row_offset + offset,
+                                }),
+                            });
+                            updated_accessor = Some(new_node);
+                        }
+                    }
+                },
+                MirAccessType::Index(index) => {
+                    let index_usize = match get_inner_const(&index) {
+                        Some(value) => value as usize,
+                        None => {
+                            self.diagnostics
+                                .diagnostic(miden_diagnostics::Severity::Error)
+                                .with_message(
+                                    "the index is not constant during constant propagation",
+                                )
+                                .with_primary_label(index.span(), "index is not constant")
+                                .emit();
+                            return Err(CompileError::Failed);
+                        },
+                    };
+                    // Check that the child node is a vector, raise diag otherwise
+                    // Replace the current node by the index-th element of the vector
+                    // Raise diag if index is out of bounds
+                    if let Op::Vector(indexable_vector) = indexable.borrow().deref() {
```

Comment:
My editor complains that this 67 line fragment is duplicated mir/src/passes/unrolling.rs:708